src/techs/techs.controller.ts

```diff
@@ -228,6 +232,133 @@ export class TechsController {
+        @Body(ValidationPipe, VoyageTeamMemberValidationPipe)
+        createTechStackCategoryDto: CreateTechStackCategoryDto,
+    ) {
+        return this.techsService.addNewTechStackCategory(
+            req,
+            createTechStackCategoryDto,
+        );
+    }
+
+    @ApiOperation({
+        summary:
+            "[Permission: own_team] Updates an existing tech stack category in the team",
+        description: "Requires login",
+    })
+    @ApiResponse({
+        status: HttpStatus.OK,
+        description: "Successfully updated a tech stack category",
+        type: TechCategoryResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.UNAUTHORIZED,
+        description: "unauthorized access - user is not logged in",
+        type: UnauthorizedErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.FORBIDDEN,
+        description: "forbidden - user does not have the required permission",
+        type: ForbiddenErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.BAD_REQUEST,
+        description: "Bad Request - tech stack category couldn't be updated",
+        type: BadRequestErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.CONFLICT,
+        description: "Tech stack category already exists for the team",
+        type: ConflictErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.NOT_FOUND,
+        description: "Invalid tech stack category id",
+        type: NotFoundErrorResponse,
+    })
+    @Patch("teams/techStackCategory")
+    updateTechStackCategory(
+        @Request() req: CustomRequest,
+        @Body(ValidationPipe)
+        updateTechStackCategoryDto: UpdateTechStackCategoryDto,
+    ) {
+        return this.techsService.updateTechStackCategory(
+            req,
+            updateTechStackCategoryDto,
+        );
+    }
+
+    @ApiOperation({
+        summary: "[Permission: own_team] Delete a team's tech stack category",
+        description: "Requires login",
+    })
+    @ApiResponse({
+        status: HttpStatus.OK,
+        description: "Tech stack category was successfully deleted",
+        type: TechCategoryDeleteResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.UNAUTHORIZED,
+        description: "unauthorized access - user is not logged in",
+        type: UnauthorizedErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.FORBIDDEN,
+        description: "forbidden - user does not have the required permission",
+        type: ForbiddenErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.BAD_REQUEST,
+        description: "Bad Request - tech stack category couldn't be deleted",
+        type: BadRequestErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.NOT_FOUND,
+        description: "Invalid tech stack category id",
+        type: NotFoundErrorResponse,
+    })
+    @Delete("teams/:teamId/techStackCategory/:techStackCategoryId")
```

Comment:
we shouldn't need teamId here as techSTackCategory is enough to identify the item we need to delete. for checking if user has the permission to do it, their teamId, teamMemberId are in `user.req`